specification/attestation/resource-manager/Microsoft.Attestation/preview/2018-09-01-preview/attestation.json

```diff
@@ -341,6 +341,80 @@
           }
         }
       }
+    },
+    "/providers/Microsoft.Attestation/defaultProviders": {
```

Comment:
>defaultProviders [](start = 38, length = 16)

With GET, the `defaultProviders` segment of the path represents a collection of resources. This requires a LIST response but you are returning a single resource, which isn't allowed. If there's truly one default provider, my recommendation would be to switch to a singleton pattern where you implement the following paths:
- GET /providers/Microsoft.Attestation/defaultProviders - returns a single resource called "default"
- GET /providers/Microsoft.Attestation/defaultProviders/default - returns the one provider. (Instead of `default` you can also name it `master`, `current`, `latest` or something similar.

If there are many default providers, then you should switch to a list response here.

Reply:
For singleton pattern, can i use `GET /providers/Microsoft.Attestation/defaultProvider` instead?

For each registered region (endpoint), there will be only one default provider.
But this resource type's RoutingType is ProxyOnly, Fanout. 
ARM will fan out and aggregate the response to all registered regions, and return a list of default providers/responses. In this case, should the response type be `AttestationProviderListResult` instead of `AttestationProvider`? 


GET /providers/Microsoft.Attestation/locations/{location}/defaultProviders returns only one default provider (each location will have only one default provider), i should switch to a singleton pattern GET /providers/Microsoft.Attestation/locations/{location}/_defaultProvider_?

Reply:
- Would default providers be returned as part of GET
​/subscriptions​/{subscriptionId}​/providers​/Microsoft.Attestation​/attestationProviders?
> Depends on subscription. 
> Default providers are created under our team's private subscription.
> For customers don't have access to our subscription, default providers will not be returned as part of GET
> ​/subscriptions​/{subscriptionId}​/providers​/Microsoft.Attestation​/attestationProviders.
> For people who have access to default provider's subscription, default providers will be returned as part of GET
> ​/subscriptions​/{subscriptionId}​/providers​/Microsoft.Attestation​/attestationProviders
>

Reply:
/providers/Microsoft.Attestation/locations/{location}/_defaultProvider_  : singleton
​/providers/Microsoft.Attestation/_defaultProviders_:   changed to list